protocol/avdl/device.avdl

```diff
@@ -7,6 +7,21 @@ protocol device {
     List devices for the user.
     */
   array<Device> deviceList(int sessionID);
+ 
+  record DeviceDetail {
+    Device device;
+    boolean eldest;
+    union {null, Device} provisioner;
```

Comment:
this is fine short term but doesn't it seem like this will ultimately be an array of typed records coming back. so like 
```
{
    type: 'Added',
    by: 'Home computer',
    when: '2016-02-25 03:23:23.238'
},
{
    type: 'Revoked',
    by: 'Home computer',
    when: '2016-02-26 03:23:23.238'
},
{
    type: 'LastActivity',
    when: '2016-02-25 03:23:23.238'
},
{
    type: 'SomethingElse',
    otherFields: 1,
    etc: 'hello'
}
```
so having all the fields be siblings and unioned w/ null could get messy right?
vs a string/enum type and a union of various types?